Plugins/AWSLambdaDeployer/Plugin.swift

```diff
@@ -0,0 +1,433 @@
+        print("Generating SAM deployment descriptor")
+        print("-------------------------------------------------------------------------")
+        
+        //
+        // Build and run the Deploy.swift package description
+        // this generates the SAM deployment descriptor
+        //
+        let deploymentDescriptorFileName = "Deploy.swift"
+        let deploymentDescriptorFilePath = "\(projectDirectory)/\(deploymentDescriptorFileName)"
+        let sharedLibraryName = "AWSLambdaDeploymentDescriptor" // provided by the swift lambda runtime
+        
+        // Check if Deploy.swift exists. Stop when it does not exist.
+        guard FileManager.default.fileExists(atPath: deploymentDescriptorFilePath) else {
+            print("`Deploy.Swift` file not found in directory \(projectDirectory)")
+            throw DeployerPluginError.deployswiftDoesNotExist
+        }
+        
+        do {
+            let cmd = [
+                swiftExecutable.string,
+                "-L \(projectDirectory)/.build/\(buildConfiguration)/",
+                "-I \(projectDirectory)/.build/\(buildConfiguration)/",
+                "-l\(sharedLibraryName)",
+                deploymentDescriptorFilePath,
+                "--archive-path", archivePath
+            ]
+            let helperCmd = cmd.joined(separator: " \\\n")
+            
+            if verboseLogging {
+                print("-------------------------------------------------------------------------")
+                print("Swift compile and run Deploy.swift")
+                print("-------------------------------------------------------------------------")
+                print("Swift command:\n\n\(helperCmd)\n")
+            }
+            
+            // create and execute a plugin helper to run the "swift" command
+            let helperFilePath = "\(projectDirectory)/compile.sh"
+            FileManager.default.createFile(atPath: helperFilePath,
+                                           contents: helperCmd.data(using: .utf8),
+                                           attributes: [.posixPermissions: 0o755])
+            let samDeploymentDescriptor = try Utils.execute(
+                executable: Path("/bin/bash"),
+                arguments: ["-c", helperFilePath],
+                customWorkingDirectory: projectDirectory,
+                logLevel: verboseLogging ? .debug : .silent)
+            // running the swift command directly from the plugin does not work 🤷‍♂️
+            //            let samDeploymentDescriptor = try execute(
+            //                executable: swiftExecutable.path,
+            //                arguments: Array(cmd.dropFirst()),
+            //                customWorkingDirectory: context.package.directory,
+            //                logLevel: configuration.verboseLogging ? .debug : .silent)
+            try FileManager.default.removeItem(atPath: helperFilePath)
```

Comment:
this probably needs to be in a defer block in case an exception is thrown before getting here

Reply:
done